packages/block-editor/src/hooks/utils.js

```diff
@@ -126,48 +126,93 @@ export function shouldSkipSerialization( blockType, featureSet, feature ) {
  * @return {Object} Settings object.
  */
 export function useBlockSettings( name, parentLayout ) {
```

Comment:
For this particular function, my goal was to actually move its call to `BlockEdit` or something and add it to the context of a block (probably as a private thing), that way it's only called once per block (as opposed to one per hook like today).